staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -865,6 +868,8 @@ type ReplicaSetCondition struct {
 	Type ReplicaSetConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=ReplicaSetConditionType"`
 	// Status of the condition, one of True, False, Unknown.
 	Status v1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,6,opt,name=lastUpdateTime"`
```

Comment:
is there a link to the proposal for this field addition? lastUpdateTime is generally not desired, because it leads to perpetual writes in steady state.

xref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties: 

> Use the LastHeartbeatTime with great caution - frequent changes to this field can cause a large fan-out effect for some resources.

also xref standard condition type definition, which intentionally does not include a field like this (https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1623-standardize-conditions/README.md) 

cc @deads2k

Reply:
Additionally, the contents of the Condition type are pretty heavily debated when changes are proposed.

IIRC there is a standard field you could add, observedGeneration. You can use that if you want to be explicit about what the controller has judged the status on.

Additionally additionally -- you can actually get this data already by examining the managed fields to see when the owner of this condition last changed it.

I doubt the arguments in favor of this field can overcome these arguments against?

/hold

Reply:
My apologies, it made sense to me when I was making the PR, but the perpetual writes are truly not desirable.
The alternatives should be sufficient as you mentioned and we should add this field only when we have a appropriate use case & KEP. I will remove it from this PR which will then significantly simplify the review

Reply:
There's no need to apologize! ... If there was a place you looked for documentation about conditions, though, let us know what it was so we can update it.

Reply:
I was looking at the Deployment condition as a reference, but I went to the docs posted above some time ago, I just forgot about / didn't revisit this ^^..